
///////////////////////////////////////////////////////////////////////////////
//
//  Copyright (c) 2008, Arizona State University
//  All rights reserved.
//  BSD License: http://www.opensource.org/licenses/bsd-license.html
//  Author: Jeff Conner
//
///////////////////////////////////////////////////////////////////////////////

///////////////////////////////////////////////////////////////////////////////
//
//  Input dialog for files.
//
///////////////////////////////////////////////////////////////////////////////

#include "GridSpaceDialog.h"

#include "Usul/Convert/Convert.h"

///////////////////////////////////////////////////////////////////////////////
//
//  Constructor.
//
///////////////////////////////////////////////////////////////////////////////

GridSpaceDialog::GridSpaceDialog ( QWidget *parent ) : BaseClass ( parent ),
_defaultValue ( 4 )
{
  // Initialize code from Designer.
  this->setupUi ( this );

  std::string defaultValue( Usul::Convert::Type< unsigned int, std::string >::convert( _defaultValue ) );

  // set the text fields with the default value
  _depth->setText( defaultValue.c_str() );

  // set the default spacing to unit length
  _xSpacing->setText( "5" );
  _ySpacing->setText( "5" );

  // set the text field for the experiment name
  spaceName->setText( "Untitled" );
 
}


///////////////////////////////////////////////////////////////////////////////
//
//  Destructor.
//
///////////////////////////////////////////////////////////////////////////////

GridSpaceDialog::~GridSpaceDialog()
{

}


///////////////////////////////////////////////////////////////////////////////
//
//  Get the expirement space name
//
///////////////////////////////////////////////////////////////////////////////

std::string GridSpaceDialog::name()
{
  std::string n ( spaceName->text().toStdString() );

  return n;
}


///////////////////////////////////////////////////////////////////////////////
//
//  Get the (x) spacing
//
///////////////////////////////////////////////////////////////////////////////

unsigned int GridSpaceDialog::x()
{
  std::string value ( _xSpacing->text().toStdString() );

  // s value
  unsigned int s ( 1 );

  if( value.length() > 0 )
  {
    s = Usul::Convert::Type< std::string, unsigned int >::convert( value );
  }

  return s;
}


///////////////////////////////////////////////////////////////////////////////
//
//  Get the (y) spacing
//
///////////////////////////////////////////////////////////////////////////////

unsigned int GridSpaceDialog::y()
{
  std::string value ( _ySpacing->text().toStdString() );

  // t value
  unsigned int t ( 1 );

  if( value.length() > 0 )
  {
    t = Usul::Convert::Type< std::string, unsigned int >::convert( value );
  }

  return t;
}


///////////////////////////////////////////////////////////////////////////////
//
//  Get the depth value (z)
//
///////////////////////////////////////////////////////////////////////////////

float GridSpaceDialog::depth()
{
  std::string value ( _depth->text().toStdString() );

  // z value
  float z ( _defaultValue );

  if( value.length() > 0 )
  {
    z = Usul::Convert::Type< std::string, float >::convert( value );
  }

  return z;

}

